### PR TITLE
General service functionality improvements to enhance the user's experience

### PR DESCRIPTION
**Pull request by bylaws**
Implements various service and kernel improvements that fix crashes in some games, most notably mmnv has been implemented, preventing a crash in SMO.

### CODE REVIEW

**Comment by bylaws** [comment]
uh ignore this for the moment, i didn't mean to create this PR yet

**Review by PixelyIon — CHANGES_REQUESTED** [review]
A few bits, really good job aside from this though

File: app/src/main/cpp/skyline/services/am/controller/IApplicationFunctions.cpp [hunk 1]
```diff
@@ -39,8 +42,15 @@ namespace skyline::service::am {
     }
 
     Result IApplicationFunctions::GetPseudoDeviceId(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
-        response.Push<u64>(0L);
-        response.Push<u64>(0L);
+        auto seedForPseudoDeviceId{state.loader->nacp->nacpContents.seedForPseudoDeviceId};
+        std::array<u8, 20> hashBuf{};
+
+        // On HOS the seed from control.ncap is hashed together with the device specific device ID seed,
+        // for us it's enough to just hash the seed from control.nacp as it provides the same guarantees.
```
[the remark this answers is not in the document]
Comment (bylaws) [reply app/src/main/cpp/skyline/services/am/controller/IApplicationFunctions.cpp:49]: Did this originally, it looks very ugly so splitting into two lines makes more sense

[the remark this answers is not in the document]
Comment (PixelyIon) [reply app/src/main/cpp/skyline/services/am/controller/IApplicationFunctions.cpp:49]: I really do dislike the break after a comma specifically, it looks really off. If you want to split at that point remove the comma and full-stop.

File: app/src/main/cpp/skyline/common/uuid.cpp [hunk 2]
```diff
@@ -74,4 +74,18 @@ namespace skyline {
 
         return uuid.Swap();
     }
+
+    UUID UUID::GenerateUuidV5(const std::array<u8, 20> sha1) {
+        constexpr u8 reserved{0x1}; // RFC4122 variant
```
[the remark this answers is not in the document]
Comment (bylaws) [reply app/src/main/cpp/skyline/common/uuid.cpp:79]: I don't do it elsewhere in the file so it's best just to leave

**Comment by bylaws** [comment]
done

**Comment by bylaws** [comment]
!!

File: app/src/main/cpp/skyline/kernel/scheduler.cpp [hunk 3]
```diff
@@ -258,6 +258,9 @@ namespace skyline::kernel {
         std::unique_lock coreLock(core->mutex);
 
         auto currentIt{std::find(core->queue.begin(), core->queue.end(), thread)}, nextIt{std::next(currentIt)};
+        if (currentIt == core->queue.end())
+            return;
+
         if (currentIt == core->queue.begin()) {
```
Comment (PixelyIon) [inline app/src/main/cpp/skyline/kernel/scheduler.cpp:264]: These are heavily correlated if statements, they should be an `if else` chain

**Review by PixelyIon — APPROVED** [review]
LGTM!

**Pull request merged**